Using dlv fails with compile errors related to _GoStringLen and _GoStringPtr

When I try to use `dlv` in `cgosqlite` or anything that depends on it, I get compiler errors:
```
$ dlv test ./
# github.com/tailscale/sqlite.test
${GOROOT}/pkg/tool/linux_amd64/link: running gcc failed: exit status 1
/usr/bin/ld: /tmp/go-link-2730833985/000000.o: in function `bind_parameter_index':
${HOME}/tailscale/sqlite/cgosqlite/cgosqlite.h:42: undefined reference to `_GoStringLen'
/usr/bin/ld: ${HOME}/tailscale/sqlite/cgosqlite/cgosqlite.h:43: undefined reference to `_GoStringPtr'
collect2: error: ld returned 1 exit status

exit status 1
```

For some reason `go test ./` still succeeds and I haven't figured out what exactly is different.

Anyway looking into this error I eventually came across golang/go#48824 and the very last comment says:
> As far as I can tell `_GoStringLen` and `_GoStringPtr` are declared in the preamble--but only if you don't use `//export`. The `static inline` trick can often work but there is no intention of officially supporting it.

So it looks like the export in `cgosqlite/cgosqlite.go` needs to be moved to a new file.

--- a/cgosqlite/cgosqlite.go
+++ b/cgosqlite/cgosqlite.go
@@ -61,24 +61,6 @@ import (
 	"github.com/tailscale/sqlite/sqliteh"
 )
 
-type walHookCb func(dbName string, pages int)
-
-var walHookFunc sync.Map // from *C.sqlite3 to walHookCb
-
-//export walCallbackGo
-func walCallbackGo(db *C.sqlite3, dbNameC *C.char, dbNameLen C.int, pages C.int) C.int {
-	v, _ := walHookFunc.Load(db)
-	hook, _ := v.(walHookCb)
-	if hook == nil {
-		return C.int(0)
-	}
-
-	dbNameB := unsafe.Slice((*byte)(unsafe.Pointer(dbNameC)), dbNameLen)
-	dbName := stringFromBytes(dbNameB)
-	hook(dbName, int(pages))
-	return C.int(0) // result's kinda useless
-}
-
 func init() {
 	C.sqlite3_initialize()
 }

--- a/cgosqlite/cgosqlite.h
+++ b/cgosqlite/cgosqlite.h
@@ -5,10 +5,6 @@
 typedef uintptr_t handle_sqlite3_stmt; // a *sqlite3_stmt
 typedef uintptr_t handle_sqlite3; // a *sqlite3 (DB conn)
 
-// Forward decls because the warnings make debugging painful.
-size_t _GoStringLen(_GoString_ s);
-const char *_GoStringPtr(_GoString_ s);
-
 // Helper methods to deal with int <-> pointer pain.
 
 static int bind_text64(handle_sqlite3_stmt stmt, int col, const char* str, sqlite3_uint64 len) {

--- a/cgosqlite/walcallback.go
+++ b/cgosqlite/walcallback.go
@@ -1,0 +1,26 @@
+package cgosqlite
+
+// #include <sqlite3.h>
+import "C"
+import (
+	"sync"
+	"unsafe"
+)
+
+type walHookCb func(dbName string, pages int)
+
+var walHookFunc sync.Map // from *C.sqlite3 to walHookCb
+
+//export walCallbackGo
+func walCallbackGo(db *C.sqlite3, dbNameC *C.char, dbNameLen C.int, pages C.int) C.int {
+	v, _ := walHookFunc.Load(db)
+	hook, _ := v.(walHookCb)
+	if hook == nil {
+		return C.int(0)
+	}
+
+	dbNameB := unsafe.Slice((*byte)(unsafe.Pointer(dbNameC)), dbNameLen)
+	dbName := stringFromBytes(dbNameB)
+	hook(dbName, int(pages))
+	return C.int(0) // result's kinda useless
+}

